tuning/utils/preprocessing_utils.py

```diff
@@ -29,8 +29,55 @@
 logger = logging.get_logger("sft_trainer_preprocessing")
 
 
+# check if the provided dataset is pretokenized or not
+# the check is taken from trl
+# https://github.com/huggingface/trl/blob/ddf4c8dc3ecf6d9ee2b24f94c62182ffd682c808/trl/trainer/sft_trainer.py#L498-L509
+def is_pretokenized_dataset(data_path: str):
+    if data_path:
+        # load one sample from the dataset in order to inspect columns
+        dataset = datasets.load_dataset("json", data_files=data_path, split="train[:1]")
+        return ("input_ids" in dataset.column_names) and (
+            "labels" in dataset.column_names
+        )
+    return False
+
+
 def validate_data_args(data_args: configs.DataArguments, packing: bool):
 
+    is_train_data_pretokenized = is_pretokenized_dataset(
+        data_path=data_args.training_data_path
+    )
+    is_eval_data_pretokenized = is_pretokenized_dataset(
+        data_path=data_args.validation_data_path
+    )
+
+    # if the provided train dataset is pretokenized
+    # however user provides formatting flags, error out
+    if is_train_data_pretokenized:
+        if (
+            data_args.response_template
+            or data_args.data_formatter_template
+            or data_args.dataset_text_field
+        ):
+            raise ValueError(
```

Comment:
Can you change this to a warning instead of an error? IMO this and packing=True should be handled the same way

Reply:
modified thanks @alex-jw-brooks